apps/explorer/src/pages/transaction-result/Events.tsx

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Disclosure } from '@headlessui/react';
+import { ChevronRight12 } from '@mysten/icons';
+import {
+    formatAddress,
+    type SuiEvent,
+    type TransactionEvents,
+} from '@mysten/sui.js';
+import clsx from 'clsx';
+import { type ReactNode } from 'react';
+
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import { CopyToClipboard } from '~/ui/CopyToClipboard';
+import { Divider } from '~/ui/Divider';
+import { ObjectLink } from '~/ui/InternalLink';
+import { Text } from '~/ui/Text';
+
+function formatTypeName(typeName: string) {
+    const split = typeName.split('<');
```

Comment:
There's some new utilities in the SDK that will make this easier to work with fwiw